### PR TITLE
Add order

### PR DESCRIPTION
**Pull request by mohamed1refaie**

### CODE REVIEW

File: src/main/java/com/piper/valley/models/repository/OrderRepository.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import com.piper.valley.models.domain.Order;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface OrderRepository extends JpaRepository<Order, Long> {
+    Optional<Order> findOneById(Long id);
```
Comment (wewark) [inline src/main/java/com/piper/valley/models/repository/OrderRepository.java]: Mine

File: src/main/java/com/piper/valley/models/service/OrderService.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import com.piper.valley.models.domain.Order;
+import com.piper.valley.models.domain.StoreProduct;
+import com.piper.valley.models.domain.User;
+
+import java.util.Collection;
+import java.util.Optional;
+
+public interface OrderService {
+   Optional<Order>getOrderById(Long id);
+   Order addOrder(User user, StoreProduct storeProduct, AddOrderForm addOrderForm);
```
Comment (wewark) [inline src/main/java/com/piper/valley/models/service/OrderService.java]: Mine

File: src/main/java/com/piper/valley/models/service/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    @Autowired
+    OrderRepository orderRepository;
+
+    @Override
+    public Optional<Order> getOrderById(Long id) {
+        return orderRepository.findOneById(id);
+    }
+
+    @Override
+    public Order addOrder(User user, StoreProduct storeProduct, AddOrderForm addOrderForm) {
```
Comment (wewark) [inline src/main/java/com/piper/valley/models/service/OrderServiceImpl.java]: Mine

File: src/main/java/com/piper/valley/models/domain/User.java [hunk 4]
```diff
@@ -132,13 +126,7 @@ public void setOrders(List<Order> orders) {
 		this.orders = orders;
 	}
 
-	public ShoppingCart getShoppingCart() {
-		return shoppingCart;
-	}
 
-	public void setShoppingCart(ShoppingCart shoppingCart) {
-		this.shoppingCart = shoppingCart;
-	}
```
Comment (wewark) [inline src/main/java/com/piper/valley/models/domain/User.java]: Mine

File: src/main/java/com/piper/valley/models/domain/ShoppingCart.java [hunk 5]
```diff
@@ -45,3 +46,4 @@ public void setOrders(List<Order> orders) {
 		this.orders = orders;
 	}
 }
+*/
```
Comment (wewark) [inline src/main/java/com/piper/valley/models/domain/ShoppingCart.java]: Mine

File: src/main/java/com/piper/valley/models/domain/Order.java [hunk 6]
```diff
@@ -14,26 +16,37 @@
 	@Column(name = "addedDate", nullable = false)
 	private Date addedDate;
 
-	@Column(name = "processedDate", nullable = false)
+	@Column(name = "processedDate")
```
Comment (wewark) [inline src/main/java/com/piper/valley/models/domain/Order.java]: Mine.
Kant nullable, what is the date of not-yet-processed orders(products in cart)?

File: src/main/java/com/piper/valley/controllers/StoreController.java [hunk 7]
```diff
@@ -142,4 +148,26 @@ public ModelAndView viewStoreProduct(@PathVariable("id") Long id) {
...
+	@RequestMapping(value = "/store/products/{id}/buy", method = RequestMethod.GET)
+	public ModelAndView addOrder(@PathVariable("id") Long id, @ModelAttribute("addOrderForm") AddOrderForm addOrderForm) {
+		Optional<StoreProduct> product = storeProductService.getProductById(id);
+		if (!product.isPresent())
+			return new ModelAndView("error/404");
+		return new ModelAndView("store/addorder", addOrderViewModel.create(addOrderForm,id));
+	}
+
+	@RequestMapping(value = "/store/products/{id}/buy", method = RequestMethod.POST)
+	public ModelAndView addOrder(@PathVariable("id") Long id,@Valid @ModelAttribute("addOrderForm")AddOrderForm addOrderForm, BindingResult bindingResult,CurrentUser currentUser,RedirectAttributes redirectAttributes)
```
Comment (wewark) [inline src/main/java/com/piper/valley/controllers/StoreController.java]: Mine

File: src/main/java/com/piper/valley/controllers/StoreController.java [hunk 8]
```diff
@@ -142,4 +148,26 @@ public ModelAndView viewStoreProduct(@PathVariable("id") Long id) {
 		return new ModelAndView("store/storeprodcutview", storeProductViewModel.create(storeProduct));
 	}
 
+	@RequestMapping(value = "/store/products/{id}/buy", method = RequestMethod.GET)
+	public ModelAndView addOrder(@PathVariable("id") Long id, @ModelAttribute("addOrderForm") AddOrderForm addOrderForm) {
```
Comment (wewark) [inline src/main/java/com/piper/valley/controllers/StoreController.java]: Mine

File: src/main/java/com/piper/valley/controllers/StoreController.java [hunk 9]
```diff
@@ -52,6 +52,12 @@
     @Autowired
     private StoreOwnerDashboardViewModel storeOwnerDashboardViewModel;
 
+    @Autowired
+	private AddOrderViewModel addOrderViewModel;
+
+    @Autowired
+	private OrderService orderService;
```
Comment (wewark) [inline src/main/java/com/piper/valley/controllers/StoreController.java]: Mine

**Comment by wewark** [comment]
3lya el 7aram amsa7 el PR kollo

**Comment by SherifAbdlNaby** [comment]
Merged with wewark's credits :'D 👍

**Pull request merged**